Core/Graphics/Source/FrameBuffer.cpp

```diff
@@ -53,37 +53,69 @@ namespace Babylon::Graphics
 
     void FrameBuffer::Clear(bgfx::Encoder& encoder, uint16_t flags, uint32_t rgba, float depth, uint8_t stencil)
     {
-        // BGFX requires us to create a new viewID, this will ensure that the view gets cleaned.
+        // BGFX requires us to create a new viewID, this will ensure that the view gets cleared.
         m_viewId = m_context.AcquireNewViewId(encoder);
 
         bgfx::setViewMode(m_viewId.value(), bgfx::ViewMode::Sequential);
         bgfx::setViewClear(m_viewId.value(), flags, rgba, depth, stencil);
         bgfx::setViewFrameBuffer(m_viewId.value(), m_handle);
 
-        // BGFX will consider the viewport when cleaning the screen, but WebGL always cleans the entire screen.
-        // That's why we always set the viewport to {0, 0, 1, 1} when cleaning.
-        bgfx::setViewRect(m_viewId.value(), 0, 0, Width(), Height());
+        // If a scissor is set, we need to set the BGFX view rect to match it before clearing.
+        // We set the view rect instead of the view scissor because BGFX clears after the view rect is set and before
+        // the view scissor is set.
+        // 
+        // If no scissor is set, WebGL cleans the entire screen, so set the view rect to cover the entire screen before
+        // clearing to match WebGL's behavior; otherwise BGFX will only clear the view rect.
+        //
+        // Note that the view rect and view scissor are reset to the desired dimensions before the encoder is submitted.
+        if (m_desiredScissor.Width != 0.0 || m_desiredScissor.Height != 0.0)
```

Comment:
```suggestion
        if (m_desiredScissor.Width != 0.0f || m_desiredScissor.Height != 0.0f)
```

Also, float comparison with exactly zero is not typical. Should this be a epsilon check?

Reply:
This is to check if the scissor is being disabled, in which case the width and height are explicitly set to 0.0. I don't think an epsilon is needed for this unless we want to start checking for invalid rects.

Reply:
Ok, since we are checking for exactly zero only for width and height, why do we need an additional check for y in the flip code?

Reply:
Ya, and I tried to document this in the comment, maybe unsuccessfully.

BGFX doesn't consider the rect to be empty if x or y is non-zero. In the case of y being non-zero, bgfx sets the top and bottom of the D3D scissor rect to y, which is what was crashing the playground on my machine yesterday.

Reply:
I think I understood your comment, maybe. Looking at this code, I don't see how scissor rect will be used if the width and height is zero. If I'm right, then bgfx will never be passed the non-zero x/y scissor values. Am I missing something?